lib/plugins/multipartBodyParser.js

```diff
@@ -116,6 +116,30 @@ function multipartBodyParser(options) {
                         if (req.params[f] && !override) {
                             return;
                         }
+                        if (Array.isArray(files[f])) {
+                            req.params[f] = [];
+                            files[f].forEach(function forEach2(file) {
+                                barrier.start('fs' + file.name);
+                                fs.readFile(file.path, function readFile(
+                                    ex,
+                                    data
+                                ) {
+                                    barrier.done('fs' + file.name);
+                                    if (ex) {
+                                        return next(
+                                            new errors.InternalError(
+                                                ex,
+                                                'unable to read file' +
+                                                    file.name
+                                            )
+                                        );
+                                    }
+                                    req.params[f].push(data);
+                                    return true;
+                                });
+                            });
+                        }
+
                         barrier.start('fs' + f);
```

Comment:
Thanks for the contribution @guoyibj!

I'm not super familiar with this piece of code. But it looks like we are trying to flatten the array here. This if condition doesn't look like it returns and the following barrier is still run against `f`. Is that intentional?

Would it be possible to flatten `files` first? I.E. convert `[f1, f2, [f3, [f4]]]` to `[f1, f2, f3, f4]` to avoid the conditional logic here?